src/pset/map/output.rs

```diff
@@ -117,13 +165,46 @@ impl Output{
         rv
     }
 
+    /// Create a output from unblinded txout
+    /// This is to be used when the txout is not blinded. This sets
+    /// the blinding key from the txout nonce
+    pub fn from_txout(txout: TxOut) -> Self {
+        let mut rv = Self::default();
+        match txout.value {
+            confidential::Value::Null => { },
+            confidential::Value::Explicit(x) => rv.amount = Some(x),
+            confidential::Value::Confidential(comm) => rv.amount_comm = Some(comm),
+        }
+        rv.script_pubkey = txout.script_pubkey;
+        match txout.asset {
+            confidential::Asset::Null => { },
+            confidential::Asset::Explicit(x) => rv.asset = Some(x),
+            confidential::Asset::Confidential(comm) => rv.asset_comm = Some(comm),
+        }
+        rv.blinding_key = txout.nonce.commitment().map(|pk| bitcoin::PublicKey {
+            key: pk,
+            compressed: true, // always serialize none as compressed pk
+        });
+        rv.value_rangeproof = txout.witness.rangeproof;
+        rv.asset_surjection_proof = txout.witness.surjection_proof;
+        rv
+    }
+
     /// Create a txout from self
     pub fn to_txout(&self) -> TxOut {
         TxOut {
-            asset: self.asset,
-            value: self.amount,
+            asset: match (self.asset_comm, self.asset) {
+                (Some(gen), _) => confidential::Asset::Confidential(gen),
+                (None, Some(id)) => confidential::Asset::Explicit(id),
+                (None, None) => confidential::Asset::Null,
+            },
+            value: match (self.amount_comm, self.amount) {
+                (Some(comm), _) => confidential::Value::Confidential(comm),
+                (None, Some(x)) => confidential::Value::Explicit(x),
+                (None, None) => confidential::Value::Null,
+            },
             nonce: self
-                .ecdh_pubkey
+                .blinding_key
```

Comment:
This should be `ecdh_pubkey` if the output is blinded and `blinding_key` if the output is unblinded.

Reply:
something along the line of ecfe679